test/test_torch.py

```diff
@@ -4885,6 +4885,131 @@ def test_diff(self, device, dtype):
+            self.assertEqual(actual_bin_edges, expected_bin_edges)
+
+        with self.assertRaisesRegex(RuntimeError, 'not supported on CPU for dtype'):
+            values = make_tensor((), device, dtype=torch.int32)
+            torch.histogram(values, 1)
+
+        inconsistent_dtype = torch.float32 if dtype != torch.float32 else torch.float64
+
+        with self.assertRaisesRegex(RuntimeError, 'input tensor and bins tensor should have same dtype'):
+            values = make_tensor((), device, dtype=dtype)
+            bins = make_tensor((), device, dtype=inconsistent_dtype)
+            torch.histogram(values, bins)
+
+        with self.assertRaisesRegex(RuntimeError, 'input tensor and weight tensor should have same dtype'):
+            values = make_tensor((), device, dtype=dtype)
+            weight = make_tensor((), device, dtype=inconsistent_dtype)
+            torch.histogram(values, 1, weight=weight)
+
+        with self.assertRaisesRegex(RuntimeError, 'input tensor and hist tensor should have same dtype'):
+            values = make_tensor((), device, dtype=dtype)
+            hist = make_tensor((), device, dtype=inconsistent_dtype)
+            bin_edges = make_tensor((), device, dtype=dtype)
+            torch.histogram(values, 1, out=(hist, bin_edges))
+
+        with self.assertRaisesRegex(RuntimeError, 'input tensor and bin_edges tensor should have same dtype'):
+            values = make_tensor((), device, dtype=dtype)
+            hist = make_tensor((), device, dtype=dtype)
+            bin_edges = make_tensor((), device, dtype=inconsistent_dtype)
+            torch.histogram(values, 1, out=(hist, bin_edges))
+
+        with self.assertRaisesRegex(RuntimeError, 'bins tensor should have dimension 1'):
+            t = make_tensor((2, 2), device, dtype=dtype)
+            torch.histogram(t, t)
+
+        with self.assertRaisesRegex(RuntimeError, 'bins tensor should have at least 1 element'):
+            t = make_tensor((0), device, dtype=dtype)
+            torch.histogram(t, t)
+
+        with self.assertRaisesRegex(RuntimeError, 'input tensor and weight tensor should have same shape'):
+            values = make_tensor((2, 2), device, dtype=dtype)
+            weight = make_tensor((1), device, dtype=dtype)
+            torch.histogram(values, 1, weight=weight)
+
+        with self.assertRaisesRegex(RuntimeError, 'hist tensor must be contiguous'):
+            values = make_tensor((), device, dtype=dtype)
+            hist = make_tensor((2), device, dtype=dtype, noncontiguous=True)
+            bin_edges = make_tensor((), device, dtype=dtype)
+            torch.histogram(values, 2, out=(hist, bin_edges))
+
+    @onlyCPU
+    @dtypes(torch.float32, torch.float64)
+    def test_histogram_noncontig(self, device, dtype):
+        # Tests handling of non-contiguous bins tensor
+        values = make_tensor((3, 5), device, dtype, low=-9, high=9)
+        bin_edges = make_tensor(10, device, dtype, low=-9, high=9).msort()[::2]
```

Comment:
Cool noncontig test, but there may be a nice way to include this in the previous test with its own "bins_noncontig" option